src/Microsoft.Diagnostics.Monitoring.WebApi/EventMonitoringPassthroughStream.cs

```diff
@@ -0,0 +1,256 @@
+                    : destinationStream,
+                bufferSize);
+        }
+
+        /// <summary>
+        /// Start processing the event stream, monitoring it for the requested event and transferring its data to the specified destination stream.
+        /// This will continue to run until the event stream is complete or a stop is requested, regardless of if the requested event has been observed.
+        /// </summary>
+        /// <param name="token">The cancellation token. It can only be signaled before processing has been started. After that point <see cref="DisposeAsync"/> should be called to stop processing.</param>
+        /// <returns></returns>
+        public Task ProcessAsync(CancellationToken token)
+        {
+            return Task.Run(() =>
+            {
+                _eventSource = new EventPipeEventSource(this);
+                _eventSource.Dynamic.AddCallbackForProviderEvent(_providerName, _eventName, TraceEventCallback);
+
+                // The EventPipeEventSource will drive the transferring of data to the destination stream as it processes events.
+                _eventSource.Process();
+            }, token);
+        }
+
+        /// <summary>
+        /// Stops monitoring for the specified stopping event. Data will continue to be written to the provided destination stream.
+        /// </summary>
+        private void StopMonitoringForEvent()
+        {
+            _eventSource?.Dynamic.RemoveCallback<TraceEvent>(TraceEventCallback);
+        }
+
+        private void TraceEventCallback(TraceEvent obj)
+        {
+            if (_payloadFilterIndexCache == null && !HydratePayloadFilterCache(obj))
+            {
+                // The payload filter doesn't map onto the actual data,
+                // we'll never match the event so stop checking it
+                // and proceed with just transferring the data to the destination stream.
+                StopMonitoringForEvent();
+                _onPayloadFilterMismatch(obj);
+                return;
+            }
+
+            if (!DoesPayloadMatch(obj))
+            {
+                return;
+            }
+
+            if (_callOnEventOnlyOnce)
+            {
+                StopMonitoringForEvent();
+            }
+
+            _onEvent(obj);
+        }
+
+        /// <summary>
+        /// Hydrates the payload filter cache.
+        /// </summary>
+        /// <param name="obj">An instance of the stopping event (matching provider, task name, and opcode), but without checking the payload yet.</param>
+        /// <returns></returns>
+        private bool HydratePayloadFilterCache(TraceEvent obj)
+        {
+            if (_payloadFilterIndexCache != null)
+            {
+                return true;
+            }
+
+            // If there's no payload filter, there's nothing to do.
+            if (_payloadFilter == null || _payloadFilter.Count == 0)
+            {
+                _payloadFilterIndexCache = new(capacity: 0);
```

Comment:
nit: Can you explicitly use the `Dictionary<int, string>` type after the `new` keyword? It's not obvious what is being constructed from this context when reading the code.